starter-code/app.js

```diff
@@ -26,6 +32,48 @@ app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
+
+// Enabling sessions
+
+app.use(session({
+  secret: "ibi-ironhack",
+  resave: true,
+  saveUninitialized: true
+}));
+
+// Setting the Strategy
+
+
+passport.serializeUser((user, callback) => {
+  callback(null, user._id);
+});
+
+passport.deserializeUser((id, callback) => {
+  User.findOne({ "_id": id }, (err, user) => {
+    if (err) { return callback(err); }
+    callback(null, user);
+  });
+});
+
+passport.use(new LocalStrategy((username, password, next) => {
+  User.findOne({ username }, (err, user) => {
+    if (err) {
+      return next(err);
+    }
+    if (!user) {
+      return next(null, false, { message: "Incorrect username" });
+    }
+    if (!bcrypt.compareSync(password, user.password)) {
+      return next(null, false, { message: "Incorrect password" });
+    }
+    return next(null, user);
+  });
+}));
+
+app.use(passport.initialize());
+app.use(passport.session());
+
+
 // Routes
 app.use("/", siteController);
```

Comment:
Notice that you are only telling app to use one router, the siteController. This means that everything you wrote and exported in admin.js and courses.js is not actually being used for routing. Also, you have double handlers for some routes such as your root url "/" which has a GET handler here and also on the admin controller. Make sure you use all the routers you need. The routes seem to be well written, but at the moment there is no access for signup for example :)